### DISCUSSION TITLE
Fix: All localbtv streams are marked wrong

### DISCUSSION THREAD

**Discussion by MapGuy11** [iptv-org/community#155] [Show and tell]
### Describe your issue

All the localbtv streams are marked Offline though they are working.

![image](https://user-images.githubusercontent.com/43685307/150040932-ebce3c1a-1af3-44c7-bf33-87a459456dc0.png)
![image](https://user-images.githubusercontent.com/43685307/150042312-cc796721-1502-4681-b68b-7314397852fb.png)
![image](https://user-images.githubusercontent.com/43685307/150042337-33c663f3-f7f9-4d5e-9d1e-80b01d255e2f.png)
![image](https://user-images.githubusercontent.com/43685307/150042381-54bf9de2-a975-4e54-beef-f7fe96d7eed5.png)

**Comment by Sphinxroot** [comment]
they work for me on kodi but  it dont star live   its act like a pvr   everytime i start the channel it restart playing the same thing over and over

**Comment by MapGuy11** [comment]
> they work for me on kodi but  it dont star live   its act like a pvr   everytime i start the channel it restart playing the same thing over and over

They do give about a 8hr buffer though your client should buffer to the front.

**Comment by what-a** [comment]
@MapGuy11 thank you for your continued effort to provide these stations! 
Streams work great when using IPTVnator and Astronica, in both Windows and Linux.
Not working in VLC which seems odd. Perhaps whatever is preventing play in VLC is also causing the bot to register them as offline.

**Comment by BellezaEmporium** [comment]
I shall examine that in further detail, but it seems that it's the "VLC" issue.

**Comment by MapGuy11** [comment]
> I shall examine that in further detail, but it seems that it's the "VLC" issue.

VLC will work just with one resolution and not the full playlist.m3u8 I have tried any way to fix it but without any luck.

**Comment by BellezaEmporium** [comment]
> > I shall examine that in further detail, but it seems that it's the "VLC" issue.
> 
> VLC will work just with one resolution and not the full playlist.m3u8 I have tried any way to fix it but without any luck.

I think that's the double redirection that he doesn't like.

Try to just give the link after.

**Comment by freearhey** [comment]
@what-a is right. The bot uses the same engine as VLC, so if the stream is not loaded in the player, it does not work for the bot either.

**Comment by MapGuy11** [comment]
> @what-a is right. The bot uses the same engine as VLC, so if the stream is not loaded in the player, it does not work for the bot either.

Oh I thought it used ffmpeg.

**Comment by MapGuy11** [comment]
> > > I shall examine that in further detail, but it seems that it's the "VLC" issue.
> > 
> > VLC will work just with one resolution and not the full playlist.m3u8 I have tried any way to fix it but without any luck.
> 
> I think that's the double redirection that he doesn't like.
> 
> Try to just give the link after.

Let me see I'll work on it later today.

**Comment by freearhey** [comment]
@MapGuy11 More precisely, the bot uses [ffprobe](https://ffmpeg.org/ffprobe.html) from the ffmpeg library to load streams. But both ffmpeg and VLC use [libavcodec](https://en.wikipedia.org/wiki/Libavcodec) to encode video and audio data.

**Comment by MapGuy11** [comment]
> @MapGuy11 More precisely, the bot uses [ffprobe](https://ffmpeg.org/ffprobe.html) from the ffmpeg library to load streams. But both ffmpeg and VLC use [libavcodec](https://en.wikipedia.org/wiki/Libavcodec) to encode video and audio data.

Interesting thanks for the insight!

**Comment by ghost** [comment]
The reason it probably doesn't work on VLC because it uses "playlist.m3u8". I checked HLSAnalyzer.com and it's a master playlist. The two other links work perfectly in VLC.

**Reply by MapGuy11** [reply]
> The reason it probably doesn't work on VLC because it uses "playlist.m3u8". I checked HLSAnalyzer.com and it's a master playlist. The two other links work perfectly in VLC.

Yes those are for the two resolutions I am trying to combine them all so it will automatically switch between each resolution based on network speed and how far away you are.

**Comment by what-a** [comment]
Even when building an adaptive m3u8 with other streams (not those from MapGuy11) I'm finding the same issue with VLC:
- An adaptive m3u8 containing only 1 absolute URL does work in VLC.
- An adaptive m3u8 containing multiple absolute URLs does **not** work in VLC.
- An adaptive m3u8 containing any number of relative URLs does work in VLC.

@MapGuy11 Could you try creating an adaptive m3u8 containing relative URLs for the resolution streams? I'm not sure that's possible due to the proxy redirect, but see if you can get that to work in any player other than VLC. If it works, then check if it appeases VLC.

**Reply by MapGuy11** [reply]
> Even when building an adaptive m3u8 with other streams (not those from MapGuy11) I'm finding the same issue with VLC:
> 
> * An adaptive m3u8 containing only 1 absolute URL does work in VLC.
> * An adaptive m3u8 containing multiple absolute URLs does **not** work in VLC.
> * An adaptive m3u8 containing any number of relative URLs does work in VLC.
> 
> @MapGuy11 Could you try creating an adaptive m3u8 containing relative URLs for the resolution streams? I'm not sure that's possible due to the proxy redirect, but see if you can get that to work in any player other than VLC. If it works, then check if it appeases VLC.

It does work with every other play than VLC that's what is annoying.

**Reply by what-a** [reply]
> > Even when building an adaptive m3u8 with other streams (not those from MapGuy11) I'm finding the same issue with VLC:
> > 
> > * An adaptive m3u8 containing only 1 absolute URL does work in VLC.
> > * An adaptive m3u8 containing multiple absolute URLs does **not** work in VLC.
> > * An adaptive m3u8 containing any number of relative URLs does work in VLC.
> > 
> > @MapGuy11 Could you try creating an adaptive m3u8 containing relative URLs for the resolution streams? I'm not sure that's possible due to the proxy redirect, but see if you can get that to work in any player other than VLC. If it works, then check if it appeases VLC.
> 
> It does work with every other play than VLC that's what is annoying.

Agreed. Are we dancing around a VLC bug?

**Comment by BellezaEmporium** [comment]
I suppose that the better option would be splitting the program into 3 scripts : 

-> PHP code that was done for getting time and other stuff.
-> Look after all available qualities and put those as m3u file and serve it somewhere
-> Generate link redirecting to the fresh m3u file.

**Reply by MapGuy11** [reply]
> I suppose that the better option would be splitting the program into 3 scripts :
> 
> -> PHP code that was done for getting time and other stuff. -> Look after all available qualities and put those as m3u file and serve it somewhere -> Generate link redirecting to the fresh m3u file.

Ok let me think about this one.

**Comment by MapGuy11** [comment]
@what-a @LaneSh4d0w All the streams with one resolution should be accessible with VLC and any other player.
Some Examples:
https://v-pi.theus6tv.tk/hls/65.3/playlist.m3u8
https://v-pi.theus6tv.tk/hls/61.4/playlist.m3u8
https://v-pi.theus6tv.tk/hls/2.2/playlist.m3u8

**Reply by BellezaEmporium** [reply]
Works on my side, VLC and HLS+DASH Player on browser

**Reply by MapGuy11** [reply]
> Works on my side, VLC and HLS+DASH Player on browser

Yeah, I am not sure how I am going to do the adaptive ones.

**Reply by BellezaEmporium** [reply]
Well, right now your implementation gives 404 on adaptives.

Well, I suppose you're using PHP, so I'll have to dig a bit, but it's doable.

**Reply by MapGuy11** [reply]
> Well, right now your implementation gives 404 on adaptives.
> 
> Well, I suppose you're using PHP, so I'll have to dig a bit, but it's doable.

They should be working just not in VLC.

**Reply by MapGuy11** [reply]
@LaneSh4d0w could you scan Indianapolis LocalBTV because there should be a 972 for Toledo but I can't find anything.

**Reply by BellezaEmporium** [reply]
I see 4 additions on my side, wait a few minutes.

**Reply by MapGuy11** [reply]
> I see 4 additions on my side, wait a few minutes.

@LaneSh4d0w Any in in.zip or I am not just seeing it in GitHub yet?

**Reply by BellezaEmporium** [reply]
not yet, it's doing some other quals

**Reply by MapGuy11** [reply]
> not yet, it's doing some other quals

@LaneSh4d0w Sounds good.

**Reply by BellezaEmporium** [reply]
1080

777 = EFFED UP (The remote server returned an error: (400) Bad Request.)
778 = EFFED UP (The remote server returned an error: (400) Bad Request.)
779 = EFFED UP (The remote server returned an error: (400) Bad Request.)
785 = EFFED UP (The remote server returned an error: (400) Bad Request.)
787 = EFFED UP (The remote server returned an error: (400) Bad Request.)
788 = EFFED UP (The remote server returned an error: (400) Bad Request.)
791 = EFFED UP (The remote server returned an error: (400) Bad Request.)
801 = EFFED UP (The remote server returned an error: (400) Bad Request.)
807 = EFFED UP (The remote server returned an error: (400) Bad Request.)
808 = EFFED UP (The remote server returned an error: (400) Bad Request.)
809 = EFFED UP (The remote server returned an error: (400) Bad Request.)
810 = EFFED UP (The remote server returned an error: (400) Bad Request.)
818 = EFFED UP (The remote server returned an error: (400) Bad Request.)
819 = EFFED UP (The remote server returned an error: (400) Bad Request.)
824 = EFFED UP (The remote server returned an error: (400) Bad Request.)
825 = EFFED UP (The remote server returned an error: (400) Bad Request.)
826 = EFFED UP (The remote server returned an error: (400) Bad Request.)
827 = EFFED UP (The remote server returned an error: (400) Bad Request.)
828 = EFFED UP (The remote server returned an error: (400) Bad Request.)
833 = EFFED UP (The remote server returned an error: (400) Bad Request.)
834 = EFFED UP (The remote server returned an error: (400) Bad Request.)
843 = EFFED UP (The remote server returned an error: (400) Bad Request.)
845 = EFFED UP (The remote server returned an error: (400) Bad Request.)
846 = EFFED UP (The remote server returned an error: (400) Bad Request.)
847 = EFFED UP (The remote server returned an error: (400) Bad Request.)
848 = EFFED UP (The remote server returned an error: (400) Bad Request.)
849 = EFFED UP (The remote server returned an error: (400) Bad Request.)
850 = EFFED UP (The remote server returned an error: (400) Bad Request.)
851 = EFFED UP (The remote server returned an error: (400) Bad Request.)
852 = EFFED UP (The remote server returned an error: (400) Bad Request.)
853 = EFFED UP (The remote server returned an error: (400) Bad Request.)
867 = EFFED UP (The remote server returned an error: (400) Bad Request.)
879 = EFFED UP (The remote server returned an error: (400) Bad Request.)
881 = EFFED UP (The remote server returned an error: (400) Bad Request.)
923 = EFFED UP (The remote server returned an error: (400) Bad Request.)
929 = EFFED UP (The remote server returned an error: (400) Bad Request.)
956 = EFFED UP (The remote server returned an error: (400) Bad Request.)
972 = EFFED UP (The remote server returned an error: (400) Bad Request.)
977 = EFFED UP (The remote server returned an error: (400) Bad Request.)

**Reply by what-a** [reply]
> @what-a @LaneSh4d0w All the streams with one resolution should be accessible with VLC and any other player. Some Examples: https://v-pi.theus6tv.tk/hls/65.3/playlist.m3u8 https://v-pi.theus6tv.tk/hls/61.4/playlist.m3u8 https://v-pi.theus6tv.tk/hls/2.2/playlist.m3u8

Yes indeed! Those are working in VLC and other players.  👍

**Reply by MapGuy11** [reply]
> > @what-a @LaneSh4d0w All the streams with one resolution should be accessible with VLC and any other player. Some Examples: https://v-pi.theus6tv.tk/hls/65.3/playlist.m3u8 https://v-pi.theus6tv.tk/hls/61.4/playlist.m3u8 https://v-pi.theus6tv.tk/hls/2.2/playlist.m3u8
> 
> Yes indeed! Those are working in VLC and other players. 👍

Yeah, so the ones that are only one resolution I can redirect with VLC liking it.

**Reply by BellezaEmporium** [reply]
Thing is I would need to reimplement the program to include everything we found.

**Reply by MapGuy11** [reply]
> Thing is I would need to reimplement the program to include everything we found.

oh ok I will look at it.

**Reply by MapGuy11** [reply]
@LaneSh4d0w @what-a  The .m3u file thing works but that seems just wrong in general I give up honestly I'll keep putting the streams out they just won't be compatible with VLC.

**Reply by what-a** [reply]
I hear ya. Seems to me this comes down to a bug in VLC and nothing more. So, I don't have a problem with the streams not being compatible in VLC. They're working flawlessly elsewhere.

**Reply by MapGuy11** [reply]
> I hear ya. Seems to me this comes down to a bug in VLC and nothing more. So, I don't have a problem with the streams not being compatible in VLC. They're working flawlessly elsewhere.

@what-a The issue is they will always be marked wrong though.

**Reply by yadzio** [reply]
Thank you guys for providing us these streams again. Will you be adding all the available channels in the future? Great work, thanks again.

**Reply by MapGuy11** [reply]
> Thank you guys for providing us these streams again. Will you be adding all the available channels in the future? Great work, thanks again.

I am working on it currently having issues with adding multiple bitrate streams so those are on pause but the subchannels I am working on. I am a one man band and getting all streams connected and available to you all takes some time. So please wait thanks :)

**Comment by what-a** [comment]
I opened a bug/issue over at the VLC camp: 
https://code.videolan.org/videolan/vlc/-/issues/26549

**Reply by MapGuy11** [reply]
> I opened a bug/issue over at the VLC camp: https://code.videolan.org/videolan/vlc/-/issues/26549

It was closed right away.

**Reply by Dum4G** [reply]
Reopened 2 hours ago

**Comment by what-a** [comment]
@MapGuy11 👏👏👏
Your commitment, patience, and perseverance to get this working is inspiring.

**Reply by MapGuy11** [reply]
> @MapGuy11 👏👏👏 Your commitment, patience, and perseverance to get this working is inspiring.

Thank You :) I try my best.

**Reply by BellezaEmporium** [reply]
> > @MapGuy11 👏👏👏 Your commitment, patience, and perseverance to get this working is inspiring.
> 
> Thank You :) I try my best.

Send me your code snippet via email. I'll check.
